apps/mobile/src/utils/localization/currency.ts

```diff
@@ -0,0 +1,80 @@
+    BRL: {code: 'BRL', flag: '🇧🇷', name: 'Real brasileiro', symbol: 'R$', position: 'before'},
+    CAD: {code: 'CAD', flag: '🇨🇦', name: 'Canadian dollar', symbol: 'CA$', position: 'before'},
+    CHF: {code: 'CHF', flag: '🇨🇭', name: 'Schweizer Franken', symbol: 'Fr', position: 'before'},
+    CLP: {code: 'CLP', flag: '🇨🇱', name: 'Peso chileno', symbol: 'CLP$', position: 'before'},
+    CNY: {code: 'CNY', flag: '🇨🇳', name: '人民币', symbol: '¥', position: 'before'},
+    CZK: {code: 'CZK', flag: '🇨🇿', name: 'Koruna česká', symbol: 'Kč', position: 'after'},
+    DKK: {code: 'DKK', flag: '🇩🇰', name: 'Danske kroner', symbol: 'kr', position: 'before'},
+    EUR: {code: 'EUR', flag: '🇪🇺', name: 'Euro', symbol: '€',position: 'before'},
+    GBP: {code: 'GBP', flag: '🇬🇧', name: 'British pound', symbol: '£', position: 'before'},
+    HKD: {code: 'HKD', flag: '🇭🇰', name: '港幣', symbol: 'HK$', position: 'before'},
+    HUF: {code: 'HUF', flag: '🇭🇺', name: 'Magyar forint', symbol: 'Ft', position: 'after'},
+    IDR: {code: 'IDR', flag: '🇮🇩', name: 'Rupiah', symbol: 'Rp', position: 'before'},
+    ILS: {code: 'ILS', flag: '🇮🇱', name: 'שקל חדש', symbol: '₪', position: 'before'},
+    INR: {code: 'INR', flag: '🇮🇳', name: 'Indian rupee', symbol: '₹', position: 'before'},
+    JPY: {code: 'JPY', flag: '🇯🇵', name: '日本円', symbol: '¥', position: 'before'},
+    KRW: {code: 'KRW', flag: '🇰🇷', name: '대한민국 원', symbol: '₩', position: 'before'},
+    MXN: {code: 'MXN', flag: '🇲🇽', name: 'Peso mexicano', symbol: 'Mex$', position: 'before'},
+    NOK: {code: 'NOK', flag: '🇳🇴', name: 'Norske kroner', symbol: 'kr', position: 'before'},
+    NZD: {code: 'NZD', flag: '🇳🇿', name: 'New Zealand dollar', symbol: 'NZ$', position: 'before'},
+    PLN: {code: 'PLN', flag: '🇵🇱', name: 'Złoty', symbol: 'zł',position: 'after'},
+    RON: {code: 'RON', flag: '🇷🇴', name: 'Leu românesc', symbol: 'lei', position: 'after'},
+    RSD: {code: 'RSD', flag: '🇷🇸', name: 'Српски динар', symbol: 'дин.', position: 'after'},
+    RUB: {code: 'RUB', flag: '🇷🇺', name: 'Российский рубль', symbol: '₽', position: 'after'},
+    SAR: {code: 'SAR', flag: '🇸🇦', name: 'ريال سعودي', symbol: 'ر.س', position: 'before'},
+    SEK: {code: 'SEK', flag: '🇸🇪', name: 'Svensk krona', symbol: 'kr', position: 'before'},
+    SGD: {code: 'SGD', flag: '🇸🇬', name: 'Singapore dollar', symbol: 'S$', position: 'before'},
+    THB: {code: 'THB', flag: '🇹🇭', name: 'Thai baht', symbol: '฿', position: 'before'},
+    TRY: {code: 'TRY', flag: '🇹🇷', name: 'Türk lirası', symbol: '₺', position: 'before'},
+    UAH: {code: 'UAH', flag: '🇺🇦', name: 'Українська гривня', symbol: '₴', position: 'after'},
+    USD: {code: 'USD', flag: '🇺🇸', name: 'US dollar', symbol: '$', position: 'before'},
+    ZAR: {code: 'ZAR', flag: '🇿🇦', name: 'South African rand', symbol: 'R', position: 'before'},
+}
+
+export function getCurrencyLeftText(currencyCode: string | undefined): string | undefined {
+    if (currencyCode === undefined) {
```

Comment:
Pro tip ⭐️. We could rewrite this to be shorter and easier to read with optional chaining operator: 

```
export function getCurrencyLeftText(currencyCode: string | undefined): string | undefined {
     return currencies[currencyCode]?.symbol;
 }
```

Reply:
And with branded types, it would look something like this: 


```
export function getCurrencyLeftText(currencyCode: Currency | undefined): string | undefined {
     if (!currencyCode) return undefined;

    const currency = currencies[currencyCode] 

    if (currency.position !== 'before') return undefined 

    return currency.symbol;
 }
```